src/gcode/gcode_to_command.cpp

```diff
@@ -360,7 +360,7 @@ void VisitCommand::to_proto_path([[maybe_unused]] const gcode::ast::M106& comman
 
     // set fan speed
     const auto set_fan_speed = std::make_shared<botcmd::FanDuty>();
-    set_fan_speed->duty = command.S.has_value() ? command.S.value() : 255;
+    set_fan_speed->duty = command.S.has_value() ? command.S.value() : 1.0;
```

Comment:
We should definitely make this change. The internal structure should be constant and be in the range $[0, 1]$. However, when the fan speed is provided it is not clear in what range it falls $[0, 1]$ or $[0, 255]$. This is context dependent and for cura generated content it will be dependent how the `machine_scale_fan_speed_zero_to_one` setting is set.

What we should do is something like the following (in addition to the already changed code)
```cpp
auto fan_speed_range = machine_scale_fan_speed_zero_to_one ? 1.0 : 255.0;
set_fan_speed->duty = command.S.has_value() ? command.S.value() / fan_speed_range : 1.0;
```
Don't think this is possible at this moment as we don't have access to the settings here. I llso this is not a big issue since we always set the `machine_scale_fan_speed_zero_to_one=true` for makerbot printers.

Reply:
@jellespijker opinions?

Reply:
many...

but no strong ones on this topic. Since we control the input we can assume we always get a value between 0 and 1.

if we want to make it a bit more robust we can add a guard for values higher then 1

if we want to make it even more robust we can also send over all the settings with the conversion script. But that introduces a whole lot of troubles, issues and pitfalls.

Just a random brainfart...
We could also send over the setting value for the check if it is between 0..1 or 0...255 by adding it to start gcode as a comment with the value of that setting. We then parse the header of comments to determine with which range we should  work

Reply:
An other option would be to differentiate the type of value based on the format : anything like `1, 5, 100, 255` would mean we are on the 0..255 range (without any decimal) and anything like `0.0, 0.15, 0.8, 1.0` would mean we are on the 0..1 range. Easy to implement but not very explicity.

However for now I would suggest letting it as is, because dulcificum is only for Method printers, and that may be true forever...